src/biconnectivity/articulation.jl

```diff
@@ -78,3 +78,60 @@ function articulation(g::AbstractGraph)
     end
     return findall(state.articulation_points)
 end
+
+function iterative_articulation(g::SimpleGraph)
+    T = eltype(g)
+    s = Vector{Tuple{T, T, T}}()
+    articulation_points = Set{T}()
+    low = zeros(T, nv(g))
+    pre = zeros(T, nv(g))
+    @inbounds for u in vertices(g)
+        pre[u] != 0 && continue
+        v = u
+        children = 0
+        wi::T = 0
```

Comment:
prefer `wi = zero(T)` here.

Reply:
I still think we should have a type annotation there, but we could also have `wi::T = zero(T)` or `wi::T = T(0)` if you prefer that. The reason is, that we later assign to `wi` in a loop. Without the type annotation that would mean, that this assignment could change the type of `wi` so hat the type is not the same in each iteration. This is really bad for performance then. 
Especially with integer types it is really easy to make this error (has also happened multiple times for me before), for example if you have an assignment `x = zero(Int8)` before a loop, but then have `x += 1` inside the loop, you will change the type of `x` to `Int`. With a type annotation that cannot happen.

Reply:
@simonschoelly this makes sense. Could you do me a favor and see where else we might be doing this inadvertently? I have a feeling you'll find a lot of cases.

Reply:
Ref #1126.